repository/src/main/java/ru/karalionak/rest/query/OrderedViewQuery.java

```diff
@@ -0,0 +1,83 @@
+package ru.karalionak.rest.query;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class OrderedViewQuery {
```

Comment:
public class OrderedViewQuery {

    private static final String QUERY_BEGINNING = "SELECT * FROM gift_certificates " +
            "JOIN gift_certificate_tag gct on gift_certificates.id = gct.certificate_id " +
            "JOIN tags on tags.id = gct.tag_id";

    private static final QueryHelper queryHelper = new QueryHelper();

    private final String query;
    private final List<Object> values;

    public OrderedViewQuery(String query, List<Object> values) {
        this.query = query;
        this.values = values;
    }

    public String getQuery() {
        return query;
    }

    public List<Object> getValues() {
        return values;
    }

    public static QueryBuilder builder() {
        return new QueryBuilder();
    }

    public static class QueryBuilder {
        private String tagName;
        private String regex;
        private SortingContext sortingContext;

        public void filteringByTagName(String tagName) {
            this.tagName = tagName;
        }

        public void filteringByNameOrDescription(String regex) {
            this.regex = regex;
        }

        public void sortingQuery(SortingContext sortingContext) {
            this.sortingContext = sortingContext;
        }

        public OrderedViewQuery build() {
            StringBuilder query = new StringBuilder(QUERY_BEGINNING);
            List<Object> values = new ArrayList<>();

            if (tagName != null || regex != null) {
                query.append(" WHERE ");
            }
            if (tagName != null) {
                query.append("(tags.tag_name=?)");
                values.add(tagName);
            }
            if (regex != null) {
                if (tagName != null) {
                    query.append(" OR ");
                }
                query.append("(certificate_name LIKE ? OR description LIKE ?)");
                values.add(regex);
            }
            if (sortingContext != null) {
                query.append(queryHelper.buildSortingQuery(sortingContext));
            }
            return new OrderedViewQuery(query.toString(), values);
        }
    }
}